### PR TITLE
🐛 Fixes pre-commit, shellcheck, curl, docs

### PR DESCRIPTION
**Pull request by mrnicegyu11**
*This is 1/3 from split huge PR https://github.com/ITISFoundation/osparc-ops-environments/pull/117*

This PR brings:
- ✨ Adds docs for running monitoring stack on local-deployment
- 🐛 Fixes minio availability check when wrong `curl` binary is in path
- 🐛 Minor: Fixes some shellcheck warnings
- 🐛 Minor: Fixes issues in custom pre-commit hook `stack-yml-generated-is-valid`

### CODE REVIEW

**Review by pcrespov — COMMENTED** [review]
I would change the commit name
:+1:

**Review by sanderegg — COMMENTED** [review]
the PR name could more focused

File: scripts/deployments/deploy.sh [hunk 1]
```diff
@@ -226,7 +226,7 @@ if [ "$start_opsstack" -eq 0 ]; then
         popd
 
         log_info "waiting for minio to run...don't worry..."
-        while [ ! "$(curl -s -o /dev/null -I -w "%{http_code}" --max-time 10 https://"${STORAGE_DOMAIN}"/minio/health/ready)" = 200 ]; do
+        while [ ! "$(/usr/bin/curl -s -o /dev/null -I -w "%{http_code}" --max-time 10 https://"${STORAGE_DOMAIN}"/minio/health/ready)" = 200 ]; do
```
[the remark this answers is not in the document]
Comment (mrnicegyu11) [reply scripts/deployments/deploy.sh:229]: true

File: scripts/pre-commit-hooks/stack-yml-generated-is-valid/launch.bash [hunk 2]
```diff
@@ -81,7 +77,7 @@ fi
 # Assert docker-copose config runs
 curl -SL --silent https://github.com/docker/compose/releases/download/v2.12.2/docker-compose-linux-x86_64 -o ./docker-compose
 chmod +x ./docker-compose
-./docker-compose --file "$repo_basedir"/stack.yml config 2>&1 | cat
+./docker-compose --file "$repo_basedir"/stack.yml config 2>&1
```
[the remark this answers is not in the document]
Comment (mrnicegyu11) [reply scripts/pre-commit-hooks/stack-yml-generated-is-valid/launch.bash:80]: Adressed in https://github.com/ITISFoundation/osparc-ops-environments/pull/124

**Pull request merged**